README.md

```diff
@@ -1 +1,415 @@
+```
+
+This command will output if the upload has been successful or not.
+
+For more information about the blob upload command see the help message:
+
+```shell
+$ azure-img-utils blob upload --help
+```
+
+### delete subcommand
+
+This subcommand allows the user to delete a blob file from the storage
+container specified. The subcommand is *azure-img-utils blob delete*.
+
+The *required* parameters for the execution of the command:
+- --storage-account
+- --blob-name
+- --container
+
+Some *optional* parameters for the execution of the command:
+- --yes  (avoids interactive confirmation for the deletion)
+
+Example:
+
+```shell
+$ azure-img-utils blob delete --storage-account myStorageAccount \
+                              --blob-name myBlobName \
+                              --container myContainerName \
+                              --yes
+```
+
+This command will output if the deletion has been successful or not.
+
+For more information about the blob delete command see the help message:
+
+```shell
+$ azure-img-utils blob delete --help
+```
+
+# API
+
+The AzureImage class can be instantiated and used as an API from code.
+This provides all the same functions as the CLI with a few additional
+helpers. For example there are waiter functions which will wait for
+a compute image to be created and/or deleted.
+
+To create an instance of AzureImage you need a *storage_account*,
+*credentials dictionary object* or *credentials file* or *sas token*
+*container* and *resource group*. Optionally you can pass
+in a Python log object and/or a *log_level* and/or a *timeout* value.
+
+Note that you can provide authentication credentials in 3 different ways:
+- with a sas_token
+- with a dictionary of credentials containing the required key values
+- with the credentials file name
+Providing just one of these options is enough to perform the authentication
+in the Azure API.
+
+```python
+azure_image = AzureImage(
+    container="my_container_name",
+    storage_account="my_storage_account",
+    credentials=my_credentials_dict,
+    credentials_file="my_credentials_file.json",
+    resource_group="my_resource_group",
+    sas_token="my_sas_token",
+    log_level=my_log_level,
+    log_callback=logger
+    timeout=myTimeout
+)
+```
+
+## Code examples
+
+With an instance of AzureImage you can perform any of the image functions
+which are available through the CLI.
+
+```python
+azure_image = AzureImage(
+    container="my_container",
+    storage_account="my_storage_account",
+    credentials_file="my_credentials.json",
+    resource_gropu="my_resource_group",
+    sas_token="my_sas_token"
+)
+```
+
+### Check if image blob exists
+```python
+blob_exists = azure_image.image_blob_exists("my_blob_name")
+```
+
+# Delete storage blob
+```python
+blob_deleted = azure_image.delete_storage_blob("my_blob_name")
+```
+
+### Upload image blob
+```python
+blob_name = azure_image.upload_image_blob(
+	"my_image_file.qcow2",
```

Comment:
Nitpick: It could be just a Github preview issue but I still see some tabbing that doesn't line up.

Reply:
yep, there were some tabs here and there :S
I've got to check my vim settings...at least to highlight visually the tabs